### PR TITLE
ID-630 Use Jersey 2 for Client, even with Spring Boot 3

### PR DESCRIPTION
**Pull request by tlangs**
Providing dependencies for the TPS client to be able to generate a client usable by other Terra services

### CODE REVIEW

**Comment by tlangs** [comment]
I published this branch locally and compiled WSM against it. Everything worked, and connected tests passed.

**Review by Shakespeared — APPROVED** [review]

**Review by sjkobori — APPROVED** [review]

File: client/src/main/resources/swaggercodegen/libraries/jersey2/JSON.mustache [hunk 1]
```diff
@@ -1,62 +0,0 @@
-package {{invokerPackage}};
```
Comment (ddietterich) [inline client/src/main/resources/swaggercodegen/libraries/jersey2/JSON.mustache:1]: Why is this deleted? Don't we need it to configure the object mapper behavior?

File: service/generators.gradle [hunk 2]
```diff
@@ -1,11 +1,9 @@
 dependencies {
-	implementation 'io.swagger.core.v3:swagger-annotations:2.2.0'
 	runtimeOnly 'org.webjars.npm:swagger-ui-dist:4.9.0'
-	swaggerCodegen 'io.swagger.codegen.v3:swagger-codegen-cli:3.0.35'
 
-	// Needed by swagger codegen
-	implementation 'javax.validation:validation-api:2.0.1.Final'
-	implementation 'javax.annotation:javax.annotation-api:1.3.2'
+	implementation 'jakarta.annotation:jakarta.annotation-api:2.1.1'
```
Comment (ddietterich) [inline service/generators.gradle:4]: nit: restore the comment?

**Review by ddietterich — COMMENTED** [review]
I don't understand what this is doing.
It looks like you have changed some of the generated to use javax, but are still requesting jakarta in the codegen gradle and including the jakarta library.

How does this work?

File: service/generators.gradle (hunk 2 above)
Comment (tlangs) [reply service/generators.gradle:4]: Will do!

File: client/src/main/resources/swaggercodegen/libraries/jersey2/JSON.mustache (hunk 1 above)
Comment (tlangs) [reply client/src/main/resources/swaggercodegen/libraries/jersey2/JSON.mustache:1]: This was added erroneously when we thought Jersey 3 would work with all the other Terra services importing the TPS client. We were wrong, so this is no longer needed.

**Comment by tlangs** [comment]
We thought we could just upgrade to Jakarta for the TPS client lib, but then, when testing against WSM, I got into library dependency hell. I think we need to upgrade all Terra services to Spring Boot 3 + Jakarta simultaneously. 

As a part of the overall migration from JavaX to Jakarta, Jakarta provides some packages under the `javax` package. The new generated code requires the new javax libs provided by jakarta for compatibility.

File: client/swagger.gradle [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
-    implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310'
-    implementation 'org.glassfish.jersey.core:jersey-client'
-    implementation 'org.glassfish.jersey.media:jersey-media-json-jackson'
-    implementation 'org.glassfish.jersey.media:jersey-media-multipart'
-    implementation 'io.swagger.core.v3:swagger-annotations'
-    swaggerCodegen 'io.swagger.codegen.v3:swagger-codegen-cli'
-    implementation "org.glassfish.jersey.inject:jersey-hk2"
-    implementation "org.glassfish.jersey.connectors:jersey-apache-connector"
+    implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.15.2'
+    implementation "org.glassfish.jersey.core:jersey-client:$jerseyVersion"
```
Comment (pshapiro4broad) [inline client/swagger.gradle:5]: Is there a reason why using the dependency management plugin doesn't pull in the correct version here? The whole point of the plugin is that you shouldn't need to hardcode versions in the gradle config.

**Comment by ddietterich** [comment]
> We thought we could just upgrade to Jakarta for the TPS client lib, but then, when testing against WSM, I got into library dependency hell. I think we need to upgrade all Terra services to Spring Boot 3 + Jakarta simultaneously.

Ah, so you already had merged code and this is the fix. That makes more sense.

File: service/generators.gradle [hunk 4]
```diff
@@ -1,11 +1,9 @@
...
+	implementation 'io.swagger.core.v3:swagger-annotations:2.2.12'
+	swaggerCodegen 'io.swagger.codegen.v3:swagger-codegen-cli:3.0.45'
```
Comment (pshapiro4broad) [inline service/generators.gradle:6]: These versions are different from those in `client/swagger.gradle`, is that intentional?

If you want to use the same versions in both, you should configure them in one of the conventions subprojects (like java-common-conventions) to avoid having to manually keep all the version numbers in sync.

File: client/swagger.gradle (hunk 3 above)
Comment (tlangs) [reply client/swagger.gradle:5]: The plugin management is pulling versions from Spring Boot 3, but we need to make our client compatible with Spring Boot 2, so we need to manually pull older versions.

File: service/generators.gradle (hunk 4 above)
Comment (tlangs) [reply service/generators.gradle:6]: The server codegen can be Spring Boot 3, but the client needs to use older versions to be compatible with other services using Spring Boot 2

Comment (ddietterich) [reply service/generators.gradle:6]: Can the Spring Boot 3 service use a Spring Boot 2 codegen?
Or is it really that we need _**everything**_ to change at once?

Comment (tlangs) [reply service/generators.gradle:6]: Yes, it can do that, we just need to provide the right generators, which is what this is doing. Its been tested with WSM and verified to work.
